### PR TITLE
Spv bloom filter

### PR DESCRIPTION
**Pull request by jamesvan2019**

### CODE REVIEW

File: common/bloom/filter_test.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+	if !bytes.Equal(got.Bytes(), want) {
+		t.Errorf("TestFilterInsertWithTweak failure: got %v want %v\n",
+			got.Bytes(), want)
+		return
+	}
+}
+
+func TestFilterBloomMatch(t *testing.T) {
+
```
Comment (dindinw) [inline common/bloom/filter_test.go:208]: why an empty test unit?

File: common/bloom/filter_test.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+	outpoint = types.NewOutPoint(hash, 0)
+
+	if f.MatchesOutPoint(outpoint) {
+		t.Errorf("TestFilterInsertUpdateNone matched outpoint %s", inputStr)
+		return
+	}
+}
+
+func TestFilterInsertP2PubKeyOnly(t *testing.T) {
+
```
Comment (dindinw) [inline common/bloom/filter_test.go:262]: empty test unit

**Review by dindinw — CHANGES_REQUESTED** [review]
Please also add an integrated test for this feature, it's similar to the `WSNotifyTest` test.

**Review by lochjin — CHANGES_REQUESTED** [review]

**Review by lochjin — CHANGES_REQUESTED** [review]

**Review by lochjin — APPROVED** [review]

**Review by lochjin — APPROVED** [review]

**Review by dindinw — APPROVED** [review]

**Comment by dindinw** [comment]
@jamesvan2019 the integrated test for SPV feature can be submit by a separated PR.

**Pull request merged**